src/dune_rules/coq_rules.ml

```diff
@@ -778,6 +778,72 @@ let setup_extraction_rules ~sctx ~dir ~dir_contents (s : Extraction.t) =
   setup_rule cctx ~dir ~sctx ~loc:s.buildable.loc ~file_targets:ml_targets
     ~source_rule coq_module
 
+let setup_ffi_rules ~sctx ~dir ({ loc; modules; library; flags } : Ffi.t) =
+  let* coqffi =
+    resolve_program sctx ~dir ~loc "coqffi" ~hint:"opam install coq-ffi"
+  in
+  let* lib =
+    Resolve.Memo.read_memo
+    @@
+    let open Resolve.Memo.O in
+    let* scope = Scope.DB.find_by_dir dir |> Resolve.Memo.lift_memo in
+    let lib_db = Scope.libs scope in
+    Lib.DB.resolve lib_db library
+  in
+  let coqffi_rule m =
+    let flags =
+      Ordered_set_lang.eval flags ~eq:( = ) ~standard:[]
```

Comment:
Did you test if this works? You're supposed to declare the `.v` files you've created in `Dir_contents`.

Reply:
I've added a test showing that coq.theory can pick up these .v files.